addons/payment_alipay/models/payment_transaction.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from werkzeug import urls
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+from odoo.addons.payment_alipay.controllers.main import AlipayController
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    # === CRUD METHODS ===#
+
+    @api.model_create_multi
+    def create(self, values_list):
+        txs = super().create(values_list)
+        for tx in txs:
+            tx._alipay_check_configuration()
+        return txs
+
+    def write(self, values):
+        res = super().write(values)
+        if values.get('currency_id') or values.get('acquirer_id'):
+            for tx in self:
+                tx._alipay_check_configuration()
+        return res
+
+    def _alipay_check_configuration(self):
+        if self.acquirer_id.alipay_payment_method == 'express_checkout'\
+                and self.currency_id.name != 'CNY':
+            raise ValidationError(
+                "Alipay: " + _(
+                    "Only transactions in Chinese Yuan (CNY) are allowed for Express Checkout."
+                )
+            )
```

Comment:
This will raise for all transactions `create`/`write` if the acquirer is not alipay and currency is not CNY, because the default value of `alipay_payment_method` is `express_checkout`, for all the acquirers

Reply:
Good point. I'll add the check back but in create/write.